src/shims/foreign_items.rs

```diff
@@ -948,12 +976,44 @@ pub trait EvalContextExt<'mir, 'tcx: 'mir>: crate::MiriEvalContextExt<'mir, 'tcx
         return Ok(None);
     }
 
-    fn eval_libc_i32(&mut self, name: &str) -> InterpResult<'tcx, i32> {
-        self
-            .eval_context_mut()
+    fn eval_libc(&mut self, name: &str) -> InterpResult<'tcx, Scalar<Tag>> {
+        self.eval_context_mut()
             .eval_path_scalar(&["libc", name])?
             .ok_or_else(|| err_unsup_format!("Path libc::{} cannot be resolved.", name).into())
-            .and_then(|scalar| scalar.to_i32())
+            .and_then(|scalar| scalar.not_undef())
+    }
+
+    fn eval_libc_i32(&mut self, name: &str) -> InterpResult<'tcx, i32> {
+        self.eval_libc(name).and_then(|scalar| scalar.to_i32())
+    }
+
+    fn set_last_error(&mut self, scalar: Scalar<Tag>) -> InterpResult<'tcx> {
+        let this = self.eval_context_mut();
+        let tcx = &{ this.tcx.tcx };
+        let errno_ptr = this.machine.last_error.unwrap();
+        this.memory_mut().get_mut(errno_ptr.alloc_id)?.write_scalar(
+            tcx,
+            errno_ptr,
+            scalar.into(),
+            Size::from_bits(32),
+        )
+    }
+
+    fn get_last_error(&mut self) -> InterpResult<'tcx, Scalar<Tag>> {
+        let this = self.eval_context_mut();
+        let tcx = &{ this.tcx.tcx };
+        let errno_ptr = this.machine.last_error.unwrap();
+        this.memory()
+            .get(errno_ptr.alloc_id)?
+            .read_scalar(tcx, errno_ptr, Size::from_bits(32))?
+            .not_undef()
+    }
+
+    fn consume_io_error(&mut self, e: std::io::Error) -> InterpResult<'tcx> {
```

Comment:
"consume" is an odd choice of words for a "set_last_error" convenience wrapper.

Is it really worth to have a helper for `set_last_error(e.raw_os_error().unwrap())`? Maybe it is, but then please call it something that indicates in its name that this sets the last_error variable.

Reply:
> This PR comes without any test... but looks like #976 some some errno things?

Yes, there is no test in this PR. But the  `fs` related functions use this when erroring. So that's why we tested that the `fs` functions failed correctly so late. Maybe I should add a comment about that in the test.

Reply:
Well I think its useful for `fs` related stuff because it allows to use the raw error itself to set `errno` instead of having to `match` the io::Error and set the proper `errno` for each case. I'll think about a decent name and add some comments explaining its behavior.